src/main/java/com/google/sps/servlets/FollowerSystemServlet.java

```diff
@@ -0,0 +1,101 @@
+ * TODO(shaargtz): join this servlet with fetchQuestions into a single question servlet.
+ */
+@WebServlet("/follower-system")
+public class FollowerSystemServlet extends HttpServlet {
+
+  /** 
+   * Executes the query to post a question to the database.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String type = request.getParameter("type");
+    int questionId = Utility.tryParseInt(request.getParameter("question-id"));
+    int userId = Utility.getUserId();
+
+    try {
+      Connection connection = DriverManager.getConnection(
+          Utility.SQL_LOCAL_URL, Utility.SQL_LOCAL_USER, Utility.SQL_LOCAL_PASSWORD);
+
+      if (type.equals("follow")) {
+        insertFollower(connection, questionId, userId);
+      } else if (type.equals("unfollow")) {
+        deleteFollower(connection, questionId, userId);
+      }
+
+    } catch (SQLException exception) {
+      // If the connection or the query don't go through, we get the log of what happened.
+      Logger logger = Logger.getLogger(FollowerSystemServlet.class.getName());
+      logger.log(Level.SEVERE, exception.getMessage(), exception);
+    }
+  }
+
+
+  /** 
+   * Makes the current user a follower of the question they clicked.
+   */
+  private void insertFollower(Connection connection, int questionId, int userId) {
+    try {
+      String insertFollowerQuery = "INSERT INTO QuestionFollower(question_id, follower_id) "
+          + "VALUES (?,?)";
+      PreparedStatement followerStatement = connection.prepareStatement(insertFollowerQuery);
+      followerStatement.setInt(SqlConstants.FOLLOWER_QUERY_QUESTIONID, questionId);
+      followerStatement.setInt(SqlConstants.FOLLOWER_QUERY_USERID, userId);
+      followerStatement.executeUpdate();
+    } catch (SQLException exception) {
+      // If the connection or the query don't go through, we get the log of what happened.
+      Logger logger = Logger.getLogger(FollowerSystemServlet.class.getName());
+      logger.log(Level.SEVERE, exception.getMessage(), exception);
+    }
+  }
+
+  /** 
+   * Makes the current user unfollow the question they clicked.
+   */
+  private void deleteFollower(Connection connection, int questionId, int userId) {
+    try {
+      String deleteFollowerQuery = "DELETE FROM QuestionFollower "
+          + "WHERE question_id=? AND follower_id=?;";
+      PreparedStatement followerStatement = connection.prepareStatement(deleteFollowerQuery);
+      followerStatement.setInt(SqlConstants.FOLLOWER_QUERY_QUESTIONID, questionId);
+      followerStatement.setInt(SqlConstants.FOLLOWER_QUERY_USERID, userId);
+      followerStatement.executeUpdate();
+    } catch (SQLException exception) {
+      // If the connection or the query don't go through, we get the log of what happened.
+      Logger logger = Logger.getLogger(FollowerSystemServlet.class.getName());
+      logger.log(Level.SEVERE, exception.getMessage(), exception);
+    }
+  }
```

Comment:
I notice the only difference between `insertFollower()` and `deleteFollower()` is the query. Maybe to not repeat so much code you could pass a boolean parameter to the function, and declare the query based on that parameter.

Reply:
nice idea, done!